opencog/openpsi/utilities.scm

```diff
@@ -326,13 +326,13 @@
 		result
 		; else check if entity is an evaluation or predicate or schema
 		(let ((type (cog-type entity)))
-			(if (or (equal? type 'PredicateNode)
-				   (equal? type 'GroundedPredicateNode)
+			(if (or (equal? type 'GroundedPredicateNode)
 				   (equal? type 'DefinedPredicateNode))
 				(set! result (cog-evaluate! (Evaluation entity (List)))))
-			(if (or (equal? type 'SchemaNode)
-            		(equal? type 'GroundedSchemaNode)
-            	    (equal? type 'DefinedSchemaeNode))
+			(if (equal? type 'PredicateNode)
+				(set! result (cog-tv entity)))
```

Comment:
>
> Should a user wanting to evaluate the value of P check for both of these
> TV's? What if they are different values?
> different values = USER ERROR ;D
>

Lets minimize the total number of checks and look-ups and if-then-else in
the logic flow.  So, if TV comes from evaluationlinks lets make it *always*
do that, instead of sometimes doing it one way and sometimes another.

Every new line of code is yet another place for a bug to sneak in. Minimize
these.

--linas


> —
>
> You are receiving this because you are subscribed to this thread.
>
> Reply to this email directly, view it on GitHub, or mute the thread.￼
>
>
>